### ISSUE TITLE
juniper_junos_software indicates successful install when installation fails

### ISSUE THREAD

**Issue by bswilden** [Juniper/ansible-junos-stdlib#309]
If the `juniper_junos_software` module only includes the `local_package` argument with `no_copy: true`, and the software package does not exist in the default location on the target device, the installation fails as expected, but the device is still rebooted, and the playbook output still indicates that the install was successful. 

For example, if the module includes only the following arguments (and there is no corresponding file in /var/tmp on the target device):
```
    juniper_junos_software:
      local_package: "software/jinstall-ppc-17.4R1.11-signed.tgz"
      no_copy: true
```

The playbook still shows a successful install:
```
TASK [Upgrade Junos OS] *********************************************************************************************************
changed: [hostname]

TASK [Print the response] *******************************************************************************************************
ok: [hostname] => {
    "response": {
        "changed": true, 
        "check_mode": false, 
        "failed": false, 
        "msg": "Package /home/user/tmp/jinstall-ppc-17.4R1.11-signed.tgz successfully installed. Reboot successfully initiated."
    }
}

PLAY RECAP *********************************************************************************************************
hostname   : ok=2    changed=1    unreachable=0    failed=0   
```

If `local_package` is included and `no_copy` is `true`, and the package does not exist on the target device, the module should probably:
1) indicate that no changes were made
2) give some error message either that the file does not exist on the target device or that `local_package` should be used with `no_copy: false`
3) skip the reboot

**Comment by stacywsmith** [comment]
What version of Junos is present on the router before this playbook is run?

**Comment by stacywsmith** [comment]
It appears this is a Junos issue specific to the MX80 platform. If I attempt to install a package which doesn't exist, then this platform produces:
```
user@mx80> request system software add /var/tmp/nonexistent.tgz | display xml 
<rpc-reply xmlns:junos="http://xml.juniper.net/junos/17.3R1/junos">
    <cli>
        <ignore-signals>
            hup
        </ignore-signals>
    </cli>
    <output>
        ERROR: package: /var/tmp/nonexistent.tgz is not found or empty
    </output>
    <cli>
        <banner></banner>
    </cli>
</rpc-reply>
```

Notice that the error message produced is NOT an RPC error, but just a text string in an `<output>` element which happens to contain the string `ERROR`. There's also no `<package-result>` element.

On a VMX running the same version of Junos, I see:
```
user@vmx> request system software add /var/tmp/nonexistent.tgz | display xml 
<rpc-reply xmlns:junos="http://xml.juniper.net/junos/17.3R1/junos">
    <cli>
        <ignore-signals>
            hup
        </ignore-signals>
    </cli>
    <output>
        /usr/libexec/ui/package: /var/tmp/nonexistent.tgz: no such file
        ERROR: No packages added
    </output>
    <package-result>1</package-result>
    <cli>
        <banner></banner>
    </cli>
</rpc-reply>
```
This platform correctly indicates a problem by the non-zero value for the `<package-result>` element.

There are other platforms which produce an RPC error upon error, and don't include a `<package-result>` element upon success. For this reason, PyEZ is considers this output to indicate that the package was successfully installed.

The bottom line is that this is either a Junos issue or a PyEZ issue. I have opened Juniper/py-junos-eznc#801 to track this.
